### PR TITLE
Fix error when using Get-ChildItem c:

### LINKED ISSUE #7032: Get-ChildItem c: throws an error.
When using Get-ChildItem with drive name without a backslash, i.e. `c:` the error is thrown.

Steps to reproduce
------------------

```powershell
Get-ChildItem c:
```

Expected behavior
-----------------

```none
Lists the content of current directory on the specified drive.
```

Actual behavior
---------------

```none
Get-ChildItem : Cannot find path 'C:\C:\' because it does not exist.
At line:1 char:1
+ Get-ChildItem c:
+ ~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\C:\:String) [Get-ChildItem], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.701
PSEdition                      Core
GitCommitId                    v6.1.0-preview.701
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by sethvs**
## PR Summary

Fix #7032, #7048 

**Get-ChildItem c: throws an error. #7032**

```powershell
Get-ChildItem c:

Get-ChildItem : Cannot find path 'C:\C:\' because it does not exist.
At line:1 char:1
+ Get-ChildItem c:
+ ~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\C:\:String) [Get-ChildItem], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand
```

**When creating file using -Path c: without a backslash, file created in the drive root and not in the current folder. #7048**

Windows PowerShell 5.1: file is created in **current** path - C:\Folder

```
[5.1] PS C:\Folder> New-Item -Path c: -Name testfile.txt


    Directory: C:\Folder


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/14/2018  12:30 PM              0 testfile.txt
```

PowerShell Core 6.0.2: file is created in **current** path - C:\Folder

```
[6.0.2] PS C:\Folder> New-Item -Path c: -Name testfile.txt


    Directory: C:\Folder


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/14/2018  12:32 PM              0 testfile.txt
```

PowerShell Core 6.1.0-preview.2: file is created in **current** path - C:\Folder

```
[6.1.0-preview.2] PS C:\Folder> New-Item -Path c: -Name testfile.txt


    Directory: C:\Folder


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/14/2018  12:28 PM              0 testfile.txt
```

PowerShell Core 6.1.0-preview.3: **file is created in root path - C:\\**

```
[6.1.0-preview.3] PS C:\Folder> New-Item -Path c: -Name testfile.txt


    Directory: C:\


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/14/2018  12:29 PM              0 testfile.txt
```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by bergmeister** [comment]
@sethvs Please push an empty commit to trigger another build since there seemed to be a sporadic problem that caused the build to hang and timeout. You could do this for example as follows, I recommend putting `[feature]` in the message so that more advanced tests run as well to test your change, which would be useful for this PR.
```
git commit -m [feature] --allow-empty
```

@iSazonov This happened to me last week as well that a build got into a hanging state. Have you see that before as well or is this probably caused by a recently added test?

**Comment by iSazonov** [comment]
@bergmeister Do you mean a problem only with CIs (not local)? 
Can you open new Issue and provide links on broken CI jobs?

**Comment by bergmeister** [comment]
@iSazonov I mean the `AppVeyor` run of this PR [here](https://ci.appveyor.com/project/PowerShell/powershell/build/v6.1.0-preview.9662). It must have got stuck or in a hanging state because it timed out after 3 hours.
The same build happened to [this ](https://ci.appveyor.com/project/PowerShell/powershell/build/v6.1.0-preview.9622) PR build of PR 7013. It seems that it happens only randomly.
I put those details into the new issue #7038

**Comment by anmenaga** [comment]
@sethvs After fixing current test failures in New-Item.Tests.ps1, please do a [feature] commit as @bergmeister suggested. Those namespaces APIs is a very fragile piece of code; so we need to run as many tests as we can to ensure that nothing gets broken by these changes.

**Comment by sethvs** [comment]
@anmenaga OK.

**Comment by sethvs** [comment]
Update-Help and PowerShellGet tests fail.
Can we restart Travis CI?

**Comment by iSazonov** [comment]
@sethvs Please look #6149 - perhaps we could fix it too.

**Comment by sethvs** [comment]
@iSazonov OK, will take a look.

**Comment by sethvs** [comment]
@adityapatwardhan @iSazonov I'm afraid I can't allocate much time at the moment. So let's split #6149 to another PR in order not to delay this one. I'm going to address it later.

**Comment by sethvs** [comment]
Empty commit to restart tests.

**Comment by adityapatwardhan** [comment]
@sethvs it seems that #7048 is fixed in preview.3

**Comment by adityapatwardhan** [comment]
@rjmholt Can you have a look too?

**Comment by sethvs** [comment]
@adityapatwardhan No, the description of this PR contains examples of the issue in 6.1.0-preview.3.

**Review by iSazonov — APPROVED** [review]
Leave a comment

File: src/System.Management.Automation/engine/SessionStateContainer.cs [hunk 1]
```diff
@@ -3774,6 +3774,11 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
                 {
                     PSTraceSource.NewArgumentNullException("paths");
                 }
+                else if (path.EndsWith((":" + Path.DirectorySeparatorChar), StringComparison.Ordinal) || path.EndsWith((":" + Path.AltDirectorySeparatorChar), StringComparison.Ordinal))
```
Comment (rjmholt) [inline src/System.Management.Automation/engine/SessionStateContainer.cs]: Might be worth hanging a line for the second path check.

**Review by rjmholt — COMMENTED** [review]
Going to build this branch and have a look at it, but it looks good so far.

File: src/System.Management.Automation/engine/SessionStateContainer.cs (hunk 1 above)
Comment (sethvs) [reply src/System.Management.Automation/engine/SessionStateContainer.cs]: Agree, done.

**Review by rjmholt — APPROVED** [review]
Ok, I've put this branch through a test run and got no unusual failures. This looks good to me!

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**